docs-en/chapter_preface/suggestions.md

```diff
@@ -2,223 +2,225 @@
 
-Compared with text, videos and pictures have a higher degree of information density and structure and are easier to understand. In this book, **key and difficult knowledge will be presented mainly in the form of animations and graphs**, while the text serves as an explanation and supplement to the animations and graphs.
+Compared with text, videos and pictures have a higher density of information and are more structured, making them easier to understand. In this book, **key and difficult concepts are mainly presented through animations and illustrations**, with text serving as explanations and supplements.
 
-If, while reading the book, you find that a particular paragraph provides an animation or a graphic solution as shown below, **please use the figure as the primary source and the text as a supplement and synthesize the two to understand the content**.
+When encountering content with animations or illustrations as shown in the figure below, **prioritize understanding the figure, with text as supplementary**, integrating both for a comprehensive understanding.
 
-![Example animation](../index.assets/animation.gif)
+![Animation Illustration Example](../index.assets/animation.gif)
 
-## Deeper Understanding In Code Practice
+## Deepen Understanding through Code Practice
 
-The companion code for this book is hosted in the [GitHub repository](https://github.com/krahets/hello-algo). As shown in the figure below, **the source code is accompanied by test samples that can be run with a single click**.
+The source code of this book is hosted on the [GitHub Repository](https://github.com/krahets/hello-algo). As shown in the figure below, **the source code comes with test examples and can be executed with just a single click**.
 
-If time permits, **it is recommended that you refer to the code and knock it through on your own**. If you have limited time to study, please read through and run all the code at least once.
+If time permits, **it's recommended to type out the code yourself**. If pressed for time, at least read and run all the codes.
 
-The process of writing code is often more rewarding than reading it. **Learning by doing is really learning**.
+Compared to just reading code, writing code often yields more learning. **Learning by doing is the real way to learn.**
 
-![Running code example](../index.assets/running_code.gif)
+![Running Code Example](../index.assets/running_code.gif)
 
-The preliminaries for running the code are divided into three main steps.
+Setting up to run the code involves three main steps.
 
-**Step 1: Install the local programming environment**. Please refer to [Appendix Tutorial](https://www.hello-algo.com/chapter_appendix/installation/) for installation, or skip this step if already installed.
+**Step 1: Install a local programming environment**. Follow the [tutorial](https://www.hello-algo.com/chapter_appendix/installation/) in the appendix for installation, or skip this step if already installed.
```

Comment:
"installation, or skip this step if already installed."
can be changed to ->
"installation instructions or proceed if already installed."